include/bouncer.h

```diff
@@ -147,11 +147,11 @@ extern int cf_sbuf_len;
 
 /*
  * Some cloud services use very long generated passwords, so give it
- * plenty of room.  Up to PostgreSQL 13, the server can handle
+ * plenty of room.  Note that, up to PostgreSQL 13, the server can handle
  * passwords up to 996 bytes, after that it's longer.  Also, libpq
  * maxes out around 1024, so going much higher is not straightforward.
```

Comment:
@petere could you point to where this limit is defined in libpq? I wasn't able to find any evidence and in testing, a password of 1150 characters worked fine.

Reply:
The relevant libpq code has been whacked around a few times since I looked into this, so the situation might have changed. Which versions did you check?

Reply:
I was looking here: https://github.com/postgres/postgres/tree/master/src/backend/libpq. I see 1024 defined [here](https://github.com/postgres/postgres/blob/master/src/backend/libpq/auth.c#L2847) but that looks like its only for RADIUS auth.

I was able to change this to 2048 and everything works fine. So that got me curious where this limit is actually being enforced

Reply:
So far as I can tell, the limit on password length was 995 bytes (1000 bytes - 4 byte packet length header - 1 byte null terminator) since 2002 (https://github.com/postgres/postgres/commit/3f63787cbfe0f1e837c92cd8ac35fd7ab811c18b#). It was then briefly changed to be unlimited (https://github.com/postgres/postgres/commit/67a472d71c98c3d2fa322a1b4013080b20720b98) and then changed again to be PG_MAX_AUTH_TOKEN_LENGTH (65536 bytes) (https://github.com/postgres/postgres/commit/9626325da5e8e23ff90091bc96535495d350f06e)

In the client code, I haven't been able to find an explicit limit. When reading passwords from a file, libpq will [enlarge its buffer in 128 byte chunks](https://github.com/postgres/postgres/blob/98e93a1fc93e9b54eb477d870ec744e9e1669f34/src/interfaces/libpq/fe-connect.c#L7080) up to INT_MAX (2147483647) . When I sent oversized passwords to my Postgres 9.6 server, the backend hung up on me.

Based on my limited reading of the source code, my guess is that the Postgres community intends to support passwords of length up to 65531 (65536 - 4 length bytes - 1 null byte) moving forward.